### DISCUSSION TITLE
Is it possible to remap the Globe/Fn key?

### DISCUSSION THREAD

**Discussion by ncbrowns** [rxhanson/Multitouch-Community#231] [General]
I'm a new MacBook Air user.  After trying a few apps, I am now using Multitouch to enable special gestures and key remaps (e.g., swapping Esc and CapsLock).

One thing I haven't been able to do is swap Fn/Globe and Control because the Fn/Globe key is not available on either of the menus to select the source and destination of a key remapping.  However, I was able to remap this key when trying Karabiner-Elements.  Would it be possible to add this option to Multitouch?

**Comment by Aaron-Rumpler** [comment]
You can do this in System Settings:
![Keyboard - Modifier Keys](https://github.com/user-attachments/assets/4b405452-7b60-4496-853a-89a3f6309a04)

**Comment by ncbrowns** [comment]
That's a good point -- I had actually done that earlier before discovering Multitouch!

I couldn't use the system settings to swap Esc and Caps Lock because they don't let you remap Esc, so those need Multitouch or a similar tool.  I guess Fn and Control have the opposite problem.  Between the two tools, I can get both swaps.

If a future version of Multitouch could handle Fn, that would let you do all the swaps in one place.

**Reply by rxhanson** [reply]
Previously I had not added this in because the fn key is a bit of an outlier on macOS - it doesn't behave the same way as the other modifier keys and doesn't appear to be remapped the same way. Additionally, many 3rd party keyboards don't execute the same fn key behavior as Apple ones, and the remapping would not take effect on those. As such, this hasn't quite bubbled back up to the top as a priority item. When I'm back in this part of the code, I'll take another look.

**Reply by ncbrowns** [reply]
Thanks for following up.  If it's straightforward to do, that would be a nice enhancement since the remappings in Multitouch would be more powerful than what the OS provides.  If it's not so simple, it's not a big deal since the OS already provides some support.   I can see Fn being problematic, particularly for third party keyboards.

After living with Fn/Globe and Control swapped (via the OS settings above) for a couple weeks, I decided to revert to the original key mappings because I'm using Fn/Globe more than Control.  So even if you added the functionality, I likely wouldn't use it now.
